### ISSUE TITLE
Page chooser should use buttons instead of links with invalid targets

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#5408]
### Issue Summary

In the page chooser, the explorer view allowing users to select a page uses links (`<a>`) to allow users to select a page:

```html
<a class="choose-page" href="#3" data-id="3" data-title="Breads" data-url="http://127.0.0.1:8000/breads/" data-parent-id="60" data-edit-url="/admin/pages/3/edit/">Breads</a>
```

There are three problems with this:

- Using `<a>` tags instead of buttons. For screen reader users, this UI will be announced as if it was a list of links for the user to navigate to those pages, not as a button to select the page. For example with VoiceOver, this would be announced as "link, Breads". We instead want "button, Select 'Breads' page" (or similar).
- Related to the above – the page title "Breads" isn’t enough context to understand what the control is for.
- Using page id as an `href`. This is problematic for accessibility testing tools, which will flag this as broken markup, since there is no element with that fragment identifier on the current page.

### Steps to Reproduce

1. Go to the CMS admin using screen-reader software
2. Open a page/link chooser

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: no

**Comment by cfarm** [comment]
Other things worth noting in this UI:

- when hovering over the page list, the whole table row changes background color, but is not itself interactive. the interactive elements in each row are:
  - the page title - clicking on this is the action the user takes to selects the link they want. therefore I'd recommend this element be a button in this UI. it also needs hover and focus styles to make it obvious that it's clickable.
  - status-tag: this element is a link to the preview of the page on the front-end. this should remain an `<a>` element. this needs additional styling to indicate the focus/hover/active status besides text color to make it more accessible.

![Screen Shot 2019-07-26 at 4 42 22 PM](https://user-images.githubusercontent.com/702526/61980385-a2b91480-afc4-11e9-80b6-3b2111db6da2.png)

**Comment by cfarm** [comment]
Additionally I would recommend *not* highlighting the whole row in this interface, since the whole row is not a clickable/interactive element

**Comment by lb-** [comment]
@thibaudcolas this is flagged as fixed in the page editor project but issue is still open. Do you know if this has been fixed?

**Comment by thibaudcolas** [comment]
@lb- we’re using this "Fixed" status to indicate existing issues we’re expecting to have fixed by the completion of the project, not the current state of issues. Right now for this I’m expecting the fix will ship in the August 2022 release.

**Comment by lb-** [comment]
Ok.  That name is a bit confusing then. But all good - maybe we should use 'planned'

**Comment by thibaudcolas** [comment]
Yeah I had a hard time thinking of a good name. "Planned" doesn’t feel right to me either – we tend to plan work in chunks of features only, while those "Fixed" issues are just there as a reference to re-check (and if all is well close them) further down the line.

**Comment by lb-** [comment]
Good call on the goal of ensuring that others do not try to pick up an issue to work on. I can't think of anything better than 'Fixed' myself. Thanks Thibaud

**Comment by lb-** [comment]
PR up for fixing this https://github.com/wagtail/wagtail/pull/9209 (note: the simplest fix, does not solve the problem of hover states for these, that will be a much bigger job and require a change of DOM structure that will add a lot of complexity).

If that PR goes through we will split this issue to a more suitable one for possibly changing the chooser approach to have an explicit 'select' button I think but that needs a larger decision.

Hopefully the PR fixes the core problem now and gives us a reference to also fix the same issue in other chooser listings.

**Comment by lb-** [comment]
Note: PR https://github.com/wagtail/wagtail/pull/9209 has been closed by me but anyone is free to pick up that PR and continue with it. It seems like this fix is not a big priority at this time but happy to help with it in the future.
